### PR TITLE
fix issue #2247; use newer kyber _R3 symbols for Botan 3.2.0 upwards

### LINKED ISSUE #2247: Fix Botan's "warning: 'Kyber1024' is deprecated"
## Description

The following warning is issued if PQC support is enabled:

```
warning: 'Kyber1024' is deprecated: Use Kyber1024_R3 [-Wdeprecated-declarations]
    Botan::KyberMode result = Botan::KyberMode::Kyber1024;
```

### PR DESCRIPTION
**Pull request by TJ-91**
fixes #2247 @ni4

### CODE REVIEW

File: src/lib/CMakeLists.txt [hunk 1]
```diff
@@ -32,7 +32,7 @@ find_package(ZLIB REQUIRED)
 # required packages
 find_package(JSON-C 0.11 REQUIRED)
 if (CRYPTO_BACKEND_BOTAN3)
-  find_package(Botan 3.0.0 REQUIRED)
+  find_package(Botan 3.2.0 REQUIRED)
```
Comment (ni4) [inline src/lib/CMakeLists.txt:35]: @TJ-91 This should be done only if PQC is enabled, for non-PQC builds we need to support Botan 3.0.0 and 3.1.1 as well.

Comment (TJ-91) [reply src/lib/CMakeLists.txt:35]: @ni4 ah, I see. I hope it's good now

**Comment by TJ-91** [comment]
@ni4 here codecov complains but I did not change the code at all aside from renaming an identifier. (And I'm not sure about the windows-msys2 test)

**Comment by ni4** [comment]
@TJ-91 codecov issue is the following one:  #2229
However, msys2 is something new which I didn't see before so filed #2274

**Comment by TJ-91** [comment]
@ni4 Can / should I change anything here?

**Comment by ni4** [comment]
> @ni4 Can / should I change anything here?

@TJ-91  Probably just force-push with some minor changes to restart the CI in hope of all become green and get merged. Not sure whether issues which cause failure would be resolved quickly.

**Comment by TJ-91** [comment]
Aside from codecov, rebasing worked.

Since the final standard version of the PQC algorithms will (most likely) be in Botan 3.6 I propose to just close this PR and get it right with the next big PR. It will change the symbols and required Botan version for PQC again and obsolete this PR. It might take a few weeks.

**Comment by ni4** [comment]
@TJ-91 Yeah, feel free to close this. Codecov stuff should be fix itself once  PR #2277 is merged.

**Pull request closed without merging**